advanced_alchemy/filters.py

```diff
@@ -329,21 +338,30 @@ def append_to_statement(
         statement: Select[tuple[ModelT]],
         model: type[ModelT],
     ) -> Select[tuple[ModelT]]:
-        field = self._get_instrumented_attr(model, self.field_name)
-        search_text = f"%{self.value}%"
-        if self.ignore_case:
-            return statement.where(field.not_ilike(search_text))
-        return statement.where(field.not_like(search_text))
+        fields = {self.field_name} if isinstance(self.field_name, str) else self.field_name
+        search_clause: list[BinaryExpression[bool]] = []
+        for field_name in fields:
+            field = self._get_instrumented_attr(model, field_name)
+            search_text = f"%{self.value}%"
+            if self.ignore_case:
+                search_clause.append(field.not_ilike(search_text))
+            else:
+                search_clause.append(field.not_like(search_text))
+        return statement.where(and_(*search_clause))
 
     def append_to_lambda_statement(
         self,
         statement: StatementLambdaElement,
         model: type[ModelT],
     ) -> StatementLambdaElement:
-        field = self._get_instrumented_attr(model, self.field_name)
-        search_text = f"%{self.value}%"
-        if self.ignore_case:
-            statement += lambda s: s.where(field.not_ilike(search_text))  # pyright: ignore[reportUnknownLambdaType,reportUnknownMemberType]
-            return statement
-        statement += lambda s: s.where(field.not_like(search_text))  # pyright: ignore[reportUnknownLambdaType,reportArgumentType,reportUnknownMemberType]
+        fields = {self.field_name} if isinstance(self.field_name, str) else self.field_name
```

Comment:
There are a couple occurrences of this, can it be made into a property called "`normalized_field_names`" that always returns a set?